1-select-tasks/22.txt

```diff
@@ -1,3 +1,20 @@
 Create a SQL query to display the name of each customer along with their most expensive purchased product:
 
-| CompanyName | ProductName | PricePerItem |
\ No newline at end of file
+| CompanyName | ProductName | PricePerItem |
+
+SELECT [c].[CompanyName], [p].[ProductName], [od].[UnitPrice] AS 'PricePerItem'
+FROM [Customers] [c]
+JOIN [Orders] [o] ON [o].[CustomerId] = [c].[CustomerId]
+JOIN [Order Details] [od] ON [o].[OrderId] = [od].[OrderId]
+JOIN [Products] [p] ON [od].[ProductId] = [p].[ProductId]
+WHERE [od].[UnitPrice] = (
+
+SELECT TOP 1 MAX([odd].[UnitPrice])
+FROM [Customers] [cc]
+JOIN [Orders] [oo] ON [oo].[CustomerId] = [cc].[CustomerId]
+JOIN [Order Details] [odd] ON [oo].[OrderId] = [odd].[OrderId]
+WHERE [cc].[CompanyName] = [c].[CompanyName]
+GROUP BY [cc].[CompanyName], [odd].[UnitPrice]
```

Comment:
why do you group by price?

Reply:
I was getting the top 1 unit price in descending order (order by) and without group by it didn't work but I reconsidered my previous task solution and redoing it